MentorMatchmaker/MentorMatchmaker/MatchMakingDomainOperations.fs

```diff
@@ -222,38 +238,309 @@ module Matchmaking =
             "
+
+        unmatchedApplicants 
+        |> List.map(fun application -> $"{dumpToFileApplicationData application}")
+        |> String.concat("\n")
+
+    let createPlannerInputs (applicants: Applicants) =
+        {   Applicants = applicants
+            ConfirmedMatches = []
+            MatchedMenteesSet = Set.empty
+            MatchedMentorSet = Set.empty
+            NumberOfHoursRequiredForOverlap = 1 }
+
+    let rec getMentorshipPairing (plannerInputs: MentorshipPlannerInputs) =
+        let filterToUnmatchedMentees (unmatchedMentees: Mentee list) (matchedMentees: Set<Mentee>) =
+            unmatchedMentees |> List.filter(fun mentee -> matchedMentees.Contains mentee <> true)
+
+        let filterToUnmatchedMentors (unmatchedMentors: Mentor list) (confirmedPairings: ConfirmedMentorshipApplication list) =
+            unmatchedMentors |> List.filter(fun unmatchedMentor -> confirmedPairings.Any(fun x -> x.MatchedMentor = unmatchedMentor) <> true)
+
+        match (plannerInputs.Applicants.Mentors, plannerInputs.NumberOfHoursRequiredForOverlap) with
+        | ([], _) ->
+            plannerInputs.ConfirmedMatches, plannerInputs
+
+        | (_, 0) ->
+            plannerInputs.ConfirmedMatches, plannerInputs
+
+        | _ ->
+            let (confirmedMatches, matchedMenteeSet, matchedMentorsSet) = getConfirmedMatchesFromPlanner plannerInputs
+            let updatedPlanner = 
+                { plannerInputs with
+                    Applicants = {
+                        Mentees = filterToUnmatchedMentees plannerInputs.Applicants.Mentees plannerInputs.MatchedMenteesSet
+                        Mentors = filterToUnmatchedMentors plannerInputs.Applicants.Mentors plannerInputs.ConfirmedMatches
+                    }
+                    ConfirmedMatches = plannerInputs.ConfirmedMatches @ confirmedMatches
+                    MatchedMenteesSet = matchedMenteeSet
+                    MatchedMentorSet = matchedMentorsSet
+                    NumberOfHoursRequiredForOverlap = plannerInputs.NumberOfHoursRequiredForOverlap - 1 }
+
+            getMentorshipPairing updatedPlanner
+    
+    let createUnmatchedApplicantPairingPermutations (mentees: UnmatchedApplicant list) (mentors: UnmatchedApplicant list) =
+        let matchPermutations = 
+            [
+                for mentor in mentors do
+                    let menteeMatches = 
+                        [
+                            for mentee in mentees do
+                                let sessionHours = generateMeetingTimes2 { CalendarSchedule.AvailableDays = mentee.AvailableDays } { CalendarSchedule.AvailableDays = mentor.AvailableDays }
+                                if mentor.EmailAddress <> mentee.EmailAddress && sessionHours.Any() then
+                                    Some {| Mentor = mentor; Mentee = mentee; SessionHours = sessionHours |}
+                                else
+                                    None
```

Comment:
Would have a TODO here about matching people while paying careful attention to their timezones here.

Reply:
The problem that we also have with timezones is that we don't have any... What we do have is merely an offset of UTC. Let's be careful with the language since we'd be talking of different things. Could be interesting to make sure that the form supports time zones instead of offsets and moving the implementation to leverage NodaTime which makes a good use of this.